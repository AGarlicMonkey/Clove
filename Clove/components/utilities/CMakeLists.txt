add_subdirectory(definitions)
add_subdirectory(delegate)
add_subdirectory(event)
add_subdirectory(expected)
add_subdirectory(log)
add_subdirectory(maths)
add_subdirectory(memory)