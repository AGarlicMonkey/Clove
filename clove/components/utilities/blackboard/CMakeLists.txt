set(SOURCE ${CMAKE_CURRENT_SOURCE_DIR}/source)
set(INCLUDE ${CMAKE_CURRENT_SOURCE_DIR}/include/Clove)

#Library
add_library(
    CloveBlackboard STATIC

	${INCLUDE}/Blackboard.hpp
	${INCLUDE}/Blackboard.inl
	${SOURCE}/Blackboard.cpp
)

#Includes
target_include_directories(
    CloveBlackboard
    
	PUBLIC
		include

)

#Libraries
target_link_libraries(
    CloveBlackboard

    PUBLIC
        CloveLog
		CloveMemory
)

if(CLOVE_BUILD_TESTS)
	add_subdirectory(tests)
endif()