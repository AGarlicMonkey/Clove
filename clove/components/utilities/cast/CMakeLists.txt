#CMake
option(CLOVE_USE_SAFE_CAST "When set to ON polyCast will dynamic_cast instead of static. Useful in debug builds to catch any errors" OFF)

#Library
add_library(
    CloveCast INTERFACE
)

#Include
target_include_directories(
	CloveCast
	
	INTERFACE 
		include
)

#Libraries
target_link_libraries(
    CloveCast

    INTERFACE
        CloveLog
)

#Preprocessor
target_compile_definitions(
	CloveCast

	INTERFACE
		CLOVE_SAFE_CAST=$<BOOL:${CLOVE_USE_SAFE_CAST}>
)