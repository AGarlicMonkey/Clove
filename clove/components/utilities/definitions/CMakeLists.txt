#CMake
option(GAR_ENABLE_DEBUG_DEFINITION "Sets the GARLIC_DEBUG macro to 1. Allowing for various debug functionality / information to be enabled" OFF)

#Library
add_library(
	CloveDefinitions INTERFACE
)

#Include
target_include_directories(
	CloveDefinitions
	
	INTERFACE 
		include
)

#Preprocessor
target_compile_definitions(
	CloveDefinitions

	INTERFACE
		#Platform
		GARLIC_PLATFORM_WINDOWS=$<PLATFORM_ID:Windows>
		GARLIC_PLATFORM_LINUX=$<PLATFORM_ID:Linux>
		GARLIC_PLATFORM_MACOS=$<PLATFORM_ID:Darwin>

		#Flags
		GARLIC_DEBUG=$<BOOL:${GAR_ENABLE_DEBUG_DEFINITION}>
)