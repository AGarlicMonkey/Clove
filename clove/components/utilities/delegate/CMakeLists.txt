set(INCLUDE ${CMAKE_CURRENT_SOURCE_DIR}/include/Clove/Delegate)
set(SOURCE ${CMAKE_CURRENT_SOURCE_DIR}/source)

#Library
add_library(
    CloveDelegate STATIC
        ${INCLUDE}/DelegateHandle.hpp
	    ${SOURCE}/DelegateHandle.cpp
	    ${INCLUDE}/MultiCastDelegate.hpp
	    ${INCLUDE}/MultiCastDelegate.inl
	    ${INCLUDE}/SingleCastDelegate.hpp
	    ${INCLUDE}/SingleCastDelegate.inl
)

#Include
target_include_directories(
	CloveDelegate
	
	PUBLIC
		include
)

if(GAR_BUILD_TESTS)
	add_subdirectory(tests)
endif()