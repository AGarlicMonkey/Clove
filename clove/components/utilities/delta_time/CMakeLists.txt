set(INCLUDE ${CMAKE_CURRENT_SOURCE_DIR}/include/Clove)
set(SOURCE ${CMAKE_CURRENT_SOURCE_DIR}/source)

#Library
add_library(
    CloveDeltaTime STATIC
        ${INCLUDE}/DeltaTime.hpp
	    ${SOURCE}/DeltaTime.cpp
)

#Include
target_include_directories(
	CloveDeltaTime
	
	PUBLIC
		include
)

copy_target_includes_to_out_dir(CloveDeltaTime)