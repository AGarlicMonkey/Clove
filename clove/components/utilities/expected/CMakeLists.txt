#Library
add_library(
    CloveExpected INTERFACE
)

#Include
target_include_directories(
	CloveExpected
	
	INTERFACE 
		include
)

if(GAR_BUILD_TESTS)
    add_subdirectory(tests)
endif()